hs-bindgen/src/HsBindgen/C/Fold/Type.hs

```diff
@@ -204,15 +205,21 @@ processTypeDecl' relPath path unit ty = case fromSimpleEnum $ cxtKind ty of
                     sizeof    <- liftIO (clang_Type_getSizeOf  ty)
                     alignment <- liftIO (clang_Type_getAlignOf ty)
 
-                    fields <- HighLevel.clang_visitChildren decl $ \cursor -> do
+                    fields' <- HighLevel.clang_visitChildren decl $ \cursor -> do
                         mfield <- mkStructField relPath unit declPath cursor
                         return $ Continue mfield
 
+                    (flam, fields) <- case partitionEithers fields' of
```

Comment:
Surprised to see that the flam does not have to be the last field.

Reply:
It has to be:

```
flam.c:2:7: error: flexible array member 'bar' with type 'int[]' is not at the end of struct
  int bar[];
      ^
flam.c:3:7: note: next field declaration is here
  int len;
      ^
1 error generated.
```

but libclang AST doesn't really enforce it. So I found it easier just to treat it as it could occur anywhere in folds, and simply rely on the fact that libclang rules out the invalid structs for us. In other words, I don't see a point to do checks that there is at most one flam at the end of a struct.

Reply:
Ah, I see, makes sense.